rfcs/0000-minimal-daemon.md

```diff
@@ -0,0 +1,71 @@
+---
+feature: minimal-daemon
+start-date: 2020-5-28
+author: John Ericson
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+`nix-daemon` should be a separate executable.
+
+# Motivation
+[motivation]: #motivation
+
+With flakes and other development, we are moving towards a more "batteries included" Nix command line.
+We don't want any of those features in the daemon, however, because the daemon is a special trusted process that we should strive to keep as simple as possible.
+\[This is comparable to a compilation pipeline, with a concise intermediate representation that nicer user-facing features "desugar" into.\]
+
+There are many things we could do about this, but I mainly want to establish some rough consensus around the problem while taking a small step to signal that consensus.
+Originally, each Nix command was its own executable, but then we combined them into one executable.
+I think this is fine for the main user-facing commands, but not good for the daemon.
+
+Finally, it's probably best not to give the daemon---a long lived process running with elevated privileges---access to tons of dead code.
```

Comment:
Can we specify exactly what privileges the daemon is running with, here? I think it's pertinent to the motivations of this RFC.

Reply:
And what is meant here by dead code?

Reply:
Will explain the "dead code". The elevated privileges thing I am not sure what to right: in theory the nix daemon just needs to be make sandboxes and access the store, but in practice we run it as root last I checked.

Reply:
Yes, it is run as Root. Currently the source doesn't allow anything else. We could check for CAPs, but we only check for Root prior to remounting the store in rw